api/oidc_test.go

```diff
@@ -0,0 +1,143 @@
+package api_test
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/buildkite/agent/v3/api"
+	"github.com/buildkite/agent/v3/logger"
+	"github.com/google/go-cmp/cmp"
+)
+
+func newOidcTokenServer(
+	t *testing.T,
+	accessToken, oidcToken, path string,
+	expectedBody []byte,
+) *httptest.Server {
+	t.Helper()
+
+	return httptest.NewServer(http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+		switch req.URL.Path {
+		case path:
+			if got, want := authToken(req), accessToken; got != want {
+				http.Error(
```

Comment:
Similarly here, and below - is it so uncomfortably long to put it on one line?

Reply:
Given that it is already quite indented, I do think that this line would be uncomfortably long if it were not wrapped.